load.go

```diff
@@ -17,17 +17,17 @@ const (
 	TextFile = "text.txt"
 )
 
-// loader is a type which loads email templates located at rootPath.
-type loader struct {
+// Loader is a type which loads email templates located at rootPath.
```

Comment:
@aeijdenberg interesting. Yeah I didn't want to export `Loader` at all (it's not needed, low API surface footprint). So even though `LoaderOpt` refers to unexported type: `type LoaderOpt func(l *loader)` is it a problem because callers will only ever be referring to `LoaderOpt` anyway. Thoughts?

Reply:
It saw it when reading the `godoc` - it's unusual to see a private class referenced in `godoc`, so I figured it was probably unusual style.

Reply:
Cool. So I was just wondering if there was a way to do this without exporting `Loader` since it's an empty struct with no methods as far as callers can see. But then I realised that there may actually be times that we want to allow the caller to set stuff on the loader. We'll leave the fields unexported for now, but imagine something like this:

```go
// WithCustomerRootPath pulls the customer template path from the request form.
// Maybe it would accept a customer struct or just a string to the path directly, this is just an example.
func WithCustomerRootPath(rootPath string, r *http.Request) emailtemplate.LoaderOpt {
	return func(l *emailtemplate.Loader) {
		l.RootPath = filepath.Join(rootPath, r.FormValue("customerTemplatePath")),
	}
}

func (w http.ResponseWriter, r *http.Request) {
	// Maybe not the best idea to load in a request anyway...
	g, err := Load(
		WithCustomerRootPath("./templates", r),
	)
}
```

Reply:
tl;dr ^ - we may as export `Loader` as you've done.